### PR TITLE
Feature/phone auth viewmodel

### PR DESCRIPTION
**Pull request by liujoshua**

### CODE REVIEW

File: android-sdk/src/main/java/org/sagebionetworks/bridge/android/viewmodel/PhoneAuthViewModel.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+                    LOGGER.warn("Sign up failed", throwable);
+
+                    signInStateLiveData.postValue(
+                            Resource.error("Error signing in: \n" + throwable.getMessage(),
+                                    new Object()));
+                }));
+    }
+
+    @Override
+    protected void onCleared() {
```
Comment (dephillipsmichael) [inline android-sdk/src/main/java/org/sagebionetworks/bridge/android/viewmodel/PhoneAuthViewModel.java:157]: do you need super.onCleared()?

File: android-sdk/src/main/java/org/sagebionetworks/bridge/android/viewmodel/PhoneAuthViewModel.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+import org.sagebionetworks.bridge.rest.model.Phone;
+import org.sagebionetworks.bridge.rest.model.UserSessionInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+
+import rx.subscriptions.CompositeSubscription;
+
+public class PhoneAuthViewModel extends ViewModel {
```
Comment (dephillipsmichael) [inline android-sdk/src/main/java/org/sagebionetworks/bridge/android/viewmodel/PhoneAuthViewModel.java:58]: Nice, this is a great improvement.  Definitely a good use-case for the ViewModel architecture.

**Review by dephillipsmichael — APPROVED** [review]

File: android-sdk/src/main/java/org/sagebionetworks/bridge/android/viewmodel/PhoneAuthViewModel.java (hunk 1 above)
Comment (liujoshua) [reply android-sdk/src/main/java/org/sagebionetworks/bridge/android/viewmodel/PhoneAuthViewModel.java:157]: probably safer, even tho super iks a no-op

File: android-sdk/src/main/java/org/sagebionetworks/bridge/android/viewmodel/PhoneAuthViewModel.java (hunk 2 above)
Comment (liujoshua) [reply android-sdk/src/main/java/org/sagebionetworks/bridge/android/viewmodel/PhoneAuthViewModel.java:58]: agreed, great suggesting

**Pull request merged**